### DISCUSSION TITLE
Custom Endpoints

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [directus/directus#6819] [Archive]
Hey, I want to display several collections in one endpoint, but unfortunately it doesn't work that way. I only get the data from the "member" collection. Can someone help me?

![image](https://user-images.githubusercontent.com/72545889/125780278-cd7b2a96-f1c7-4deb-80c4-1b45308074c6.png)

**Comment by SeanDylanGoff** [comment]
With res.json you end the request... You need to wait for both results (Promise.all) and then merge into one Object and res.json that.

Have you seen the GraphQL endpoint though? There you can easily fetch 2 or more collections in one request...

**Reply by Lucas-Gruber** [reply]
Sorry but i don't use GraphQL. Is there any way to do this with Get requests?

**Reply by Lucas-Gruber** [reply]
So it doesn't have to be an endpoint, but I was told on the Directus Discord server there is the possibility with endpoints and hooks, but I was not helped with how to do this.

**Reply by SeanDylanGoff** [reply]
Either you send two get requests, or use graphql or make a custom endpoint. Your endpoint will work if you use Promise.all (if you don't know how read about how promises and async exactly work)

**Reply by Lucas-Gruber** [reply]
can you give me the code please? i'm sorry but i'm not very familiar with js, and i don't know how it works. The example from the Docs doesn't help me that much either.

ps: i would like to do it with more collections than just two, that was just one example. and I can only integrate the api via get requests.

**Reply by SeanDylanGoff** [reply]
Send your code in a code block here (three backticks on a line then the code then again three on a new line) and ill fix it. But i'd really read about promises and await as you'll get much further then

**Reply by Lucas-Gruber** [reply]
Okay thank you! This is the Code:
```js
module.exports = function registerEndpoint(router, { services, exceptions }) {
	const { ItemsService } = services;
	const { ServiceUnavailableException } = exceptions;

	router.get('/', (req, res, next) => {
		const memberService = new ItemsService('member', { schema: req.schema, accountability: req.accountability });
    const gameplayService = new ItemsService('gameplays', { schema: req.schema, accountability: req.accountability });

		memberService
			.readByQuery({ fields: ['name','titel','potrait','mitglied','rollen'] })
			.then((results) => (gameplayService))
      .catch((error) => {
        return next(new ServiceUnavailableException(error.message));
      });
    
    gameplayService
			.readByQuery({ fields: ['gameplay'] })
			.then((results) => res.json(results))
      .catch((error) => {
        return next(new ServiceUnavailableException(error.message));
      });
      
	});
};
```
And yes, unfortunately I just need a quick solution, I can do a better one afterwards :)

**Comment by benhaynes** [comment]
Perhaps I'm misunderstanding the problem, but you can fetch data from multiple collections if that days is joined relationally. For instance, if you're trying to get gameplay data from each member, you can use dot notation on the `fields` parameter:

https://docs.directus.io/reference/api/query/#fields

**Reply by Lucas-Gruber** [reply]
no sorry, i will all data from gameplays and members and other collections, with one request.

**Reply by Lucas-Gruber** [reply]
sorry but i don't understand what i have to do. my goal is to be able to display all members, gameplays and other collections on an overview page, for example. (with all fields)

**Reply by benhaynes** [reply]
You can do this out of the box by just making two rest calls. If you need to have this data through one endpoint, then I would look into the answer above.

**Reply by Lucas-Gruber** [reply]
Okay, can you please give me an example from the answer above? 😅

**Reply by Lucas-Gruber** [reply]
What you meaning?

**Reply by benhaynes** [reply]
Hey Thomas, this is not premium support, this is just a place where you can ask questions. Hopefully the community will give you some guidance, otherwise you'll have to research and debug this on your own. You can also try community support at https://directus.chat

**Reply by Lucas-Gruber** [reply]
I know it's not premium support, and I only asked a question, but if I'm only answered with emojis, I could do without an answer.
